Source/DafnyCore/DafnyOptions.cs

```diff
@@ -333,6 +333,9 @@ public enum IncludesModes {
     public bool AuditProgram = false;
 
     public static string DefaultZ3Version = "4.12.1";
+    // Not directly user-configurable, only recorded once we discover it
+    public string SolverIdentifier { get; private set; }
```

Comment:
It was added in https://github.com/dafny-lang/dafny/pull/3663/files#diff-ba72c3c1efed223d4216f91a154cd44c8e31134d76911321a965d27db8eeb8acR1119, and it looks like it was just a case of "hmm getting the version is relatively expensive, we should probably cache this safely in case it's called a lot". So I'm not too worried about a multi-solver workflow showing up, but I'll make `SetZ3Options` throw if invoked multiple times just in case.